lib/Street.js

```diff
@@ -22,7 +22,10 @@ Street.prototype.setId = function( id ){
 Street.prototype.getNames = function(){ return this.names; };
 Street.prototype.setNames = function( names ){
   if( !Array.isArray( names ) || !names.length ){ throw new Error( 'invalid names array' ); }
-  this.names = names;
+  this.names = names.map( function( name ){
+    if( 'string' !== typeof name ){ throw new Error( 'invalid name string' ); }
```

Comment:
sorry, I don't understand the question?

line 24 throws on error, I don't see why throwing on line 26 would be any different?